src/api/routers/login_router.py

```
@@ -31,20 +36,81 @@ async def login(
+    print(store_token)
+    if store_token.decode("utf-8") == jti_of_token:
+        raise HTTPException(
+            status_code=422,
+            detail="Old Token",
+        )
+
+    store_jti(redis, current_user, jti_of_token)
     new_access_token = Authorize.create_access_token(subject=current_user)
     new_refresh_token = Authorize.create_refresh_token(subject=current_user)
     return Token(
         access_token=new_access_token,
         refresh_token=new_refresh_token,
         token_type="Bearer",
     )
+
+
+@login_router.post("/sent-reset-link")
+async def sent_reset_link(
+    email: str, session: AsyncSession = Depends(get_db), redis=Depends(get_redis_client)
+):
+    user = await UserCRUD.get_user_by_email(email, session)
+    if user:
+        try:
+            reset_token = generate_reset_password_token()
+            store_reset_token(redis, email, reset_token)
+            send_reset_token(email, reset_token)
+
+            return {"message": "The token has been sent"}
+        except Exception as e:
+            return {"error": f"Error: {str(e)}"}
+
+    else:
+        raise HTTPException(
+            status_code=404,
+            detail="Email has not found",
+        )
+
+
+@login_router.post("/reset-password")
+async def reset_password(
+    email: str,
+    token: str,
+    new_password: str,
+    confirm_password: str,
+    session: AsyncSession = Depends(get_db),
+    redis=Depends(get_redis_client),
+):
+    try:
+        store_token = get_reset_token(redis, email)
+        if store_token == token and new_password == confirm_password:
+            updated_user = await UserCRUD.change_user_password(
+                email, new_password, session
+            )
+
+            if updated_user:
+                redis.delete(email)
+                return {"message": "password has been changed"}
+            else:
+                return {"message": "error change"}
+        else:
+            return {"message": "error"}
```

you don't want to actually raise an Error and have an fastapi ExceptionHandler?

---

oh, i'm sorry, just missed it, didn't remove the stoppers. ofc, will change it